In [1]:
import sys
from pathlib import Path
import os
import json

sys.path.append(str(Path.cwd().parent.parent))
print(str(Path.cwd().parent.parent))
print("Current working directory: ", os.getcwd())

from ModelDevelopment.CentralPipeline import CentralPipeline
from ModelDevelopment.ImageBatchPipeline import ImageBatchPipeline
from DataProcessing.DataPreProcessing import DataPaths

c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition
Current working directory:  c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\ModelDevelopment\experiments


In [2]:
# gt_test = DataPaths.CHALLENGE_DATA_GT.value
# #Schema of gt_test: {"0": 4, "1": 93, "2": 25, "3": -1, "4": -1,

# # load the json and compile a list of all tracklets with -1 as not_legible and all other tracklets as legible
# with open(gt_test, 'r') as f:
#     data = json.load(f)
#     legible = []
#     illegible = []
    
#     for key, value in data.items():
#         if value == -1:
#             illegible.append(key)
#         else:
#             legible.append(key)
    
# print("Legible tracklets: ", legible)
# print("Illegible tracklets: ", illegible)

In [3]:
# # Delete soccer_ball.json files
# import os
# root_path = r"C:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\test\images"

# for tracklet in range(1, 1210):
#   path = os.path.join(root_path, str(tracklet))
  
#   # If there is a .soccer_ball.json file in there, delete it
#   if os.path.exists(os.path.join(path, "soccer_ball.json")):
#     os.remove(os.path.join(path, "soccer_ball.json"))

In [4]:
tracklets_to_process_override_batch1 = [i for i in range(0, 712)] # COLIN
tracklets_to_process_override_batch2 = [i for i in range(712, 1425)] # JARED Second half of the tracklets (605 items)

pipeline = CentralPipeline(
  #tracklets_to_process_override=[109, 88, 34, 151, 19, 68, 67, 128, 57, 178, 189, 196, 214, 242, 258, 271, 304, 327, 320, 352, 375, 378, 414, 416, 439, 425, 451, 453, 454, 466, 499, 511, 513, 526, 528, 567, 576, 594, 619, 623, 646, 687, 682, 709, 711, 717, 739, 758, 767, 795, 788, 840, 843, 832, 841, 852, 847, 851, 856, 850, 868, 866, 871, 901, 902, 904, 900, 907, 928, 939, 944, 961, 984, 983, 1005, 1039, 1064, 1063, 1077, 1091, 1101, 1137, 1152, 1153, 1160, 1167, 1157, 1165, 1206],
  #tracklets_to_process_override=[327, 711, 907, 1064], # These ones are now passing
  #tracklets_to_process_override=[327]
  #tracklets_to_process_override=[0],
  #tracklets_to_process_override=illegible,
  #tracklets_to_process_override=[168, 167, 162, 97, 39, 87, 89, 134, 61],
  #tracklets_to_process_override=[168, 167, 162, 97, 39, 87, 89, 134, 61, 172, 164, 199, 215, 260, 267, 269, 281, 276, 288, 295, 317, 325, 340, 345, 374, 394, 388, 386, 391, 406, 408, 434, 437, 459, 471, 514, 532, 558, 683, 748, 770, 790, 802, 827, 861, 869, 879, 897, 895, 909, 964, 981, 999, 1023, 1031, 1025, 1033, 1030, 1055, 1088, 1087, 1169, 1175, 1174, 1173, 1183, 1170, 1179, 1184, 1185],
  #num_tracklets=1210,
  tracklets_to_process_override=tracklets_to_process_override_batch1,
  #tracklets_to_process_override=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
  #num_images_per_tracklet=50,
  input_data_path=DataPaths.CHALLENGE_DATA_DIR.value,
  output_processed_data_path=DataPaths.PROCESSED_DATA_OUTPUT_DIR_CHALLENGE.value,
  common_processed_data_dir=DataPaths.COMMON_PROCESSED_OUTPUT_DATA_CHALLENGE.value,
  gt_data_path="",#DataPaths.TEST_DATA_GT.value,
  display_transformed_image_sample=False, # NOTE: DO NOT USE. Code is parallelized so we cannot show images anymore. Code breaks, but first one will show if True.
  use_cache=True, # Set to false to rebuild the cache
  suppress_logging=False,
  use_image_enhancement=True,
  
  # --- PARALLELIZATION PARAMS --- These settings are optimal for an NVIDIA RTX 3070 Ti Laptop GPU.
  num_workers=2,            # CRITICAL optimisation param. Adjust accordingly. Pre-processing: 4 | STR: 2 | Crops: 4
  tracklet_batch_size=32,   # CRITICAL optimisation param. Adjust accordingly. Always 32
  image_batch_size=100,     # CRITICAL optimisation param. Adjust accordingly. Pre-processing: 200 | STR: 100 | Crops: 200
  num_threads_multiplier=5  # CRITICAL optimisation param. Adjust accordingly. Pre-processing: 4
  )                         #                              Interpretation: num_threads = num_workers * num_threads_multiplier

2025-03-31 10:15:23 [INFO] DataPreProcessing initialized. Universe of available data paths:
2025-03-31 10:15:23 [INFO] ROOT_DATA_DIR: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted
2025-03-31 10:15:23 [INFO] TEST_DATA_GT: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\test\test_gt.json
2025-03-31 10:15:23 [INFO] TRAIN_DATA_GT: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\train\train_gt.json
2025-03-31 10:15:23 [INFO] TEST_DATA_DIR: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\test\images
2025-03-31 10:15:23 [INFO] TRAIN_DATA_DIR: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\train\images
2025-03-31 10:15:23 [INFO] CHALLENGE_DATA_DIR: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\extracted\challenge\images
2

In [ ]:
pipeline.run_soccernet(
  run_soccer_ball_filter=False,
  generate_features=True,
  run_filter=True,
  run_legible=True,
  run_legible_eval=False,
  run_pose=True,
  run_crops=True,
  run_str=True,
  run_combine=False,
  run_eval=False,
  use_clip4str=True,
  pyscript=False,
  )

2025-03-31 10:15:23 [INFO] Running the SoccerNet pipeline.
2025-03-31 10:15:23 [INFO] Tracklet override applied. Using provided tracklets: 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 1

  0%|          | 0/23 [00:00<?, ?it/s]

2025-03-31 10:15:23 [INFO] Tracklet batch size: 32
2025-03-31 10:15:23 [INFO] Image batch size: 100
2025-03-31 10:15:23 [INFO] Number of workers: 2
2025-03-31 10:15:23 [INFO] Number of threads created: 10
2025-03-31 10:15:23 [INFO] All tracklets in 0-32 are cached. Skipping feature generation.
2025-03-31 10:15:23 [INFO] Skipping first 18 cached tracklets in batch 32-64. Processing the rest.
2025-03-31 10:15:23 [INFO] Using double parallelization: multiprocessing + CUDA batch processing.


Processing tracklets (CUDA + CPU):   0%|          | 0/14 [00:00<?, ?it/s]

2025-03-31 10:15:48 [INFO] Creating placeholder data files for Soccer Ball Filter.


Processing Batch Tracklets (32-64):   0%|          | 0/14 [00:00<?, ?it/s]

c:\Users\colin\miniconda3\envs\UBC\Lib\site-packages\pytorch_lightning\utilities\migration\migration.py:208: You have multiple `ModelCheckpoint` callback states in this checkpoint, but we found state keys that would end up colliding with each other after an upgrade, which means we can't differentiate which of your checkpoint callbacks needs which states. At least one of your `ModelCheckpoint` callbacks will not be able to reload the state.
Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:15:51 [INFO] Saved features for tracklet with shape (716, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\50\features.npy
2025-03-31 10:15:51 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:15:53 [INFO] Saved features for tracklet with shape (733, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\52\features.npy
2025-03-31 10:15:53 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:15:55 [INFO] Saved features for tracklet with shape (715, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\54\features.npy
2025-03-31 10:15:55 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:15:57 [INFO] Saved features for tracklet with shape (750, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\57\features.npy
2025-03-31 10:15:57 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:15:59 [INFO] Saved features for tracklet with shape (750, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\58\features.npy
2025-03-31 10:15:59 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:16:02 [INFO] Saved features for tracklet with shape (532, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\51\features.npy
2025-03-31 10:16:02 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:16:03 [INFO] 2025-03-31 10:16:02 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:16:02 [INFO] Round 1: Excluded 3 outliers
2025-03-31 10:16:02 [INFO] Round 1: Average score: 0.9981070756912231
2025-03-31 10:16:02 [INFO] Excluded images: ['50_714.jpg', '50_715.jpg', '50_716.jpg']

2025-03-31 10:16:03 [ERROR] 
2025-03-31 10:16:03 [INFO] Done removing outliers
2025-03-31 10:16:03 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:16:03 [INFO] Classifying legibility of image(s) using pre-trained model.


c:\Users\colin\miniconda3\envs\UBC\Lib\site-packages\torchvision\models\_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
  warnings.warn(
c:\Users\colin\miniconda3\envs\UBC\Lib\site-packages\torchvision\models\_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=ResNet50_Weights.IMAGENET1K_V1`. You can also use `weights=ResNet50_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)


2025-03-31 10:16:06 [INFO] 2025-03-31 10:16:05 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:16:05 [INFO] Round 1: Excluded 7 outliers
2025-03-31 10:16:05 [INFO] Round 1: Average score: 0.9972158670425415
2025-03-31 10:16:05 [INFO] Excluded images: ['52_565.jpg', '52_566.jpg', '52_584.jpg', '52_746.jpg', '52_747.jpg', '52_748.jpg', '52_749.jpg']

2025-03-31 10:16:06 [ERROR] 
2025-03-31 10:16:06 [INFO] Done removing outliers
2025-03-31 10:16:06 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:16:06 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 10:16:07 [INFO] Saved features for tracklet with shape (436, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\55\features.npy
2025-03-31 10:16:07 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:16:09 [INFO] 2025-03-31 10:16:08 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:16:08 [INFO] Round 1: Excluded 10 outliers
2025-03-31 10:16:08 [INFO] Round 1: Average score: 0.9964770078659058
2025-03-31 10:16:08 [INFO] Excluded images: ['54_107.jpg', '54_109.jpg', '54_273.jpg', '54_274.jpg', '54_275.jpg', '54_645.jpg', '54_712.jpg', '54_713.jpg', '54_714.jpg', '54_715.jpg']

2025-03-31 10:16:09 [ERROR] 
2025-03-31 10:16:09 [INFO] Done removing outliers
2025-03-31 10:16:09 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:16:09 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 10:16:11 [INFO] 2025-03-31 10:16:10 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:16:10 [INFO] Round 1: Excluded 1 outliers
2025-03-31 10:16:10 [INFO] Round 1: Average score: 0.9986622929573059
2025-03-31 10:16:10 [INFO] Excluded images: ['57_717.jpg']

2025-03-31 10:16:11 [ERROR] 
2025-03-31 10:16:11 [INFO] Done removing outliers

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:16:14 [INFO] 2025-03-31 10:16:13 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:16:13 [INFO] Round 1: Excluded 55 outliers
2025-03-31 10:16:13 [INFO] Round 1: Average score: 0.986057460308075
2025-03-31 10:16:13 [INFO] Excluded images: ['58_304.jpg', '58_305.jpg', '58_306.jpg', '58_307.jpg', '58_308.jpg', '58_309.jpg', '58_310.jpg', '58_311.jpg', '58_312.jpg', '58_316.jpg', '58_317.jpg', '58_318.jpg', '58_472.jpg', '58_473.jpg', '58_474.jpg', '58_499.jpg', '58_501.jpg', '58_502.jpg', '58_524.jpg', '58_525.jpg', '58_526.jpg', '58_527.jpg', '58_528.jpg', '58_529.jpg', '58_530.jpg', '58_531.jpg', '58_532.jpg', '58_533.jpg', '58_534.jpg', '58_535.jpg', '58_536.jpg', '58_537.jpg', '58_538.jpg', '58_539.jpg', '58_540.jpg', '58_541.jpg', '58_542.jpg', '58_543.jpg', '58_544.jpg', '58_545.jpg', '58_547.jpg', '58_568.jpg', '58_569.jpg', '58_570.jpg', '58_640.jpg', '58_641.jpg', '58_666.jpg', '58_667.jpg', '58_668.jpg', '58_669.jpg', '58_670.jpg', '58_694.jpg', '58_695.j

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:16:22 [INFO] Saved features for tracklet with shape (601, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\53\features.npy
2025-03-31 10:16:22 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file
2025-03-31 10:16:22 [INFO] 2025-03-31 10:16:21 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:16:21 [INFO] Round 1: Excluded 36 outliers
2025-03-31 10:16:21 [INFO] Round 1: Average score: 0.9808850884437561
2025-03-31 10:16:21 [INFO] Excluded images: ['55_301.jpg', '55_304.jpg', '55_305.jpg', '55_306.jpg', '55_307.jpg', '55_308.jpg', '55_489.jpg', '55_490.jpg', '55_491.jpg', '55_492.jpg', '55_493.jpg', '55_494.jpg', '55_495.jpg', '55_496.jpg', '55_497.jpg', '55_498.jpg', '55_513.jpg', '55_514.jpg', '55_515.jpg', '55_516.jpg', '55_517.jpg', '55_80.jpg', '55_81.jpg', '55_82.jpg', '55_83.jpg', '55_84.jpg', '55_85.jpg', '55_86.jpg', '55_87.jpg', '55_88.jpg', '55_89.jp

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:16:44 [INFO] Saved features for tracklet with shape (750, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\60\features.npy
2025-03-31 10:16:45 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file
2025-03-31 10:16:45 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.441020667552948, 0.4392718994600743, 0.4639602303504944
2025-03-31 10:16:45 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.0, 0.0, 0.0
2025-03-31 10:16:45 [INFO] Tracklet 50 is illegible.
2025-03-31 10:16:45 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\50\legible_results.json
2025-03-31 10:16:45 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\50\legible_results.json
2025-03-31 10

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:16:52 [INFO] Saved features for tracklet with shape (750, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\61\features.npy
2025-03-31 10:16:52 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file
2025-03-31 10:16:55 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.5936472713947296, 0.5926325026458288, 0.606975793838501
2025-03-31 10:16:55 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5936472713947296, 0.5926325026458288, 0.606975793838501
2025-03-31 10:16:55 [INFO] Tracklet 52 is legible.
2025-03-31 10:16:55 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\52\legible_results.json
2025-03-31 10:16:55 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\chall

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:16:59 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.4617415964603424, 0.46030562435481565, 0.5002191066741943
2025-03-31 10:16:59 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5002191066741943, 0.5002191066741943, 0.5002191066741943
2025-03-31 10:16:59 [INFO] Tracklet 54 is legible.
2025-03-31 10:16:59 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\54\legible_results.json
2025-03-31 10:16:59 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\54\legible_results.json
2025-03-31 10:16:59 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\54\illegible_results.json
2025-03-31 10:16:59 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\J

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:17:05 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.5296118855476379, 0.5258753817451023, 0.5595659613609314
2025-03-31 10:17:05 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5307410359382629, 0.5286960556707012, 0.5595659613609314
2025-03-31 10:17:05 [INFO] Tracklet 51 is legible.
2025-03-31 10:17:05 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\51\legible_results.json
2025-03-31 10:17:05 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\51\legible_results.json
2025-03-31 10:17:05 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\51\illegible_results.json
2025-03-31 10:17:05 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Je

Processing tracklets (CUDA + CPU):   0%|          | 0/32 [00:00<?, ?it/s]

2025-03-31 10:18:22 [INFO] Creating placeholder data files for Soccer Ball Filter.


Processing Batch Tracklets (64-96):   0%|          | 0/32 [00:00<?, ?it/s]

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:18:25 [INFO] Saved features for tracklet with shape (750, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\64\features.npy
2025-03-31 10:18:25 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:18:27 [INFO] Saved features for tracklet with shape (750, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\66\features.npy
2025-03-31 10:18:27 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:18:30 [INFO] Saved features for tracklet with shape (750, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\68\features.npy
2025-03-31 10:18:30 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:18:32 [INFO] Saved features for tracklet with shape (251, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\71\features.npy
2025-03-31 10:18:32 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:18:34 [INFO] Saved features for tracklet with shape (580, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\72\features.npy
2025-03-31 10:18:34 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:18:35 [INFO] 2025-03-31 10:18:34 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:18:34 [INFO] Round 1: Excluded 62 outliers
2025-03-31 10:18:34 [INFO] Round 1: Average score: 0.9907081723213196
2025-03-31 10:18:34 [INFO] Excluded images: ['64_181.jpg', '64_182.jpg', '64_183.jpg', '64_184.jpg', '64_316.jpg', '64_317.jpg', '64_318.jpg', '64_319.jpg', '64_320.jpg', '64_321.jpg', '64_322.jpg', '64_365.jpg', '64_366.jpg', '64_367.jpg', '64_368.jpg', '64_369.jpg', '64_370.jpg', '64_371.jpg', '64_388.jpg', '64_389.jpg', '64_390.jpg', '64_391.jpg', '64_392.jpg', '64_393.jpg', '64_394.jpg', '64_395.jpg', '64_396.jpg', '64_397.jpg', '64_398.jpg', '64_399.jpg', '64_400.jpg', '64_401.jpg', '64_402.jpg', '64_403.jpg', '64_481.jpg', '64_483.jpg', '64_495.jpg', '64_497.jpg', '64_498.jpg', '64_499.jpg', '64_500.jpg', '64_501.jpg', '64_502.jpg', '64_503.jpg', '64_504.jpg', '64_505.jpg', '64_506.jpg', '64_512.jpg', '64_513.jpg', '64_514.jpg', '64_516.jpg', '64_518.jpg', '64_519.

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:18:41 [INFO] 2025-03-31 10:18:40 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:18:40 [INFO] Round 1: Excluded 82 outliers
2025-03-31 10:18:40 [INFO] Round 1: Average score: 0.9871559143066406
2025-03-31 10:18:40 [INFO] Excluded images: ['68_10.jpg', '68_102.jpg', '68_105.jpg', '68_107.jpg', '68_108.jpg', '68_109.jpg', '68_11.jpg', '68_110.jpg', '68_111.jpg', '68_112.jpg', '68_113.jpg', '68_114.jpg', '68_115.jpg', '68_116.jpg', '68_117.jpg', '68_118.jpg', '68_119.jpg', '68_12.jpg', '68_120.jpg', '68_121.jpg', '68_122.jpg', '68_123.jpg', '68_124.jpg', '68_125.jpg', '68_126.jpg', '68_127.jpg', '68_128.jpg', '68_129.jpg', '68_13.jpg', '68_134.jpg', '68_136.jpg', '68_137.jpg', '68_138.jpg', '68_139.jpg', '68_14.jpg', '68_140.jpg', '68_141.jpg', '68_142.jpg', '68_143.jpg', '68_144.jpg', '68_145.jpg', '68_148.jpg', '68_149.jpg', '68_15.jpg', '68_150.jpg', '68_151.jpg', '68_152.jpg', '68_153.jpg', '68_154.jpg', '68_155.jpg', '68_5.jpg', '68_501.jpg', '68_502.jpg', '6

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`
Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:18:44 [INFO] 2025-03-31 10:18:43 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:18:43 [INFO] Round 1: Excluded 2 outliers
2025-03-31 10:18:43 [INFO] Round 1: Average score: 0.9964966773986816
2025-03-31 10:18:43 [INFO] Excluded images: ['71_500.jpg', '71_501.jpg']

2025-03-31 10:18:44 [ERROR] 
2025-03-31 10:18:44 [INFO] Done removing outliers
2025-03-31 10:18:44 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:18:44 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 10:18:46 [INFO] Saved features for tracklet with shape (750, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\67\features.npy
2025-03-31 10:18:46 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file
2025-03-31 10:18:48 [INFO] 2025-03-31 10:18:46 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:18:46 [INFO] Round 1: Excluded 52 outliers
2025

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:18:53 [INFO] 2025-03-31 10:18:51 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:18:51 [INFO] Round 1: Excluded 19 outliers
2025-03-31 10:18:51 [INFO] Round 1: Average score: 0.9960711002349854
2025-03-31 10:18:51 [INFO] Excluded images: ['65_109.jpg', '65_115.jpg', '65_116.jpg', '65_517.jpg', '65_518.jpg', '65_524.jpg', '65_525.jpg', '65_535.jpg', '65_537.jpg', '65_543.jpg', '65_570.jpg', '65_575.jpg', '65_578.jpg', '65_605.jpg', '65_609.jpg', '65_610.jpg', '65_611.jpg', '65_620.jpg', '65_621.jpg']

2025-03-31 10:18:53 [ERROR] 
2025-03-31 10:18:53 [INFO] Done removing outliers
2025-03-31 10:18:53 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:18:53 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 10:18:55 [INFO] Saved features for tracklet with shape (750, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\70\features.npy
2025-03-31 10:18:55 [IN

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:19:02 [INFO] Saved features for tracklet with shape (589, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\74\features.npy
2025-03-31 10:19:02 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file
2025-03-31 10:19:03 [INFO] 2025-03-31 10:19:02 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:19:02 [INFO] Round 1: Excluded 93 outliers
2025-03-31 10:19:02 [INFO] Round 1: Average score: 0.9830573797225952
2025-03-31 10:19:02 [INFO] Excluded images: ['67_216.jpg', '67_217.jpg', '67_218.jpg', '67_219.jpg', '67_220.jpg', '67_221.jpg', '67_222.jpg', '67_223.jpg', '67_224.jpg', '67_500.jpg', '67_501.jpg', '67_502.jpg', '67_503.jpg', '67_504.jpg', '67_525.jpg', '67_526.jpg', '67_527.jpg', '67_528.jpg', '67_529.jpg', '67_531.jpg', '67_533.jpg', '67_541.jpg', '67_554.jpg', '67_556.jpg', '67_557.jpg', '67_558.jpg', '67_559.jpg', '67_560.jpg', '67_561.jpg', '67_562.jpg', 

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:19:18 [INFO] 2025-03-31 10:19:16 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:19:16 [INFO] Round 1: Excluded 31 outliers
2025-03-31 10:19:16 [INFO] Round 1: Average score: 0.9882418513298035
2025-03-31 10:19:16 [INFO] Excluded images: ['74_194.jpg', '74_195.jpg', '74_197.jpg', '74_275.jpg', '74_276.jpg', '74_277.jpg', '74_278.jpg', '74_279.jpg', '74_280.jpg', '74_382.jpg', '74_384.jpg', '74_385.jpg', '74_466.jpg', '74_467.jpg', '74_468.jpg', '74_469.jpg', '74_470.jpg', '74_471.jpg', '74_472.jpg', '74_473.jpg', '74_569.jpg', '74_570.jpg', '74_571.jpg', '74_572.jpg', '74_573.jpg', '74_574.jpg', '74_579.jpg', '74_580.jpg', '74_586.jpg', '74_587.jpg', '74_589.jpg']

2025-03-31 10:19:18 [ERROR] 
2025-03-31 10:19:18 [INFO] Done removing outliers
2025-03-31 10:19:18 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:19:18 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 10:19:21 [INFO] Saved features for tracklet with sh

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:19:26 [INFO] Saved features for tracklet with shape (703, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\76\features.npy
2025-03-31 10:19:26 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.5274321734905243, 0.5286195732340842, 0.5769268274307251
2025-03-31 10:19:26 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file
2025-03-31 10:19:26 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.528902143239975, 0.5315418412064433, 0.5769268274307251
2025-03-31 10:19:26 [INFO] Tracklet 68 is legible.
2025-03-31 10:19:26 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\68\legible_results.json
2025-03-31 10:19:26 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\chal

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:19:30 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.46410737931728363, 0.46368815727306134, 0.4937503933906555
2025-03-31 10:19:30 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.0, 0.0, 0.0
2025-03-31 10:19:30 [INFO] Tracklet 72 is illegible.
2025-03-31 10:19:30 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\72\legible_results.json
2025-03-31 10:19:30 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\72\legible_results.json
2025-03-31 10:19:30 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\72\illegible_results.json
2025-03-31 10:19:30 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\je

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:19:36 [INFO] 2025-03-31 10:19:35 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:19:35 [INFO] Round 1: Excluded 4 outliers
2025-03-31 10:19:35 [INFO] Round 1: Average score: 0.9965587258338928
2025-03-31 10:19:35 [INFO] Excluded images: ['75_175.jpg', '75_578.jpg', '75_579.jpg', '75_580.jpg']

2025-03-31 10:19:36 [ERROR] 
2025-03-31 10:19:36 [INFO] Done removing outliers
2025-03-31 10:19:36 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:19:36 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 10:19:37 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.5254024267196655, 0.5235085730350458, 0.5971003770828247
2025-03-31 10:19:37 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.526646614074707, 0.5282304512367476, 0.5971003770828247
2025-03-31 10:19:37 [INFO] Tracklet 65 is legible.
2025-03-31 10:19:37 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Reco

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:19:42 [INFO] 2025-03-31 10:19:41 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:19:41 [INFO] Round 1: Excluded 21 outliers
2025-03-31 10:19:41 [INFO] Round 1: Average score: 0.9949303269386292
2025-03-31 10:19:41 [INFO] Excluded images: ['76_161.jpg', '76_162.jpg', '76_174.jpg', '76_185.jpg', '76_187.jpg', '76_246.jpg', '76_247.jpg', '76_248.jpg', '76_271.jpg', '76_272.jpg', '76_353.jpg', '76_405.jpg', '76_436.jpg', '76_443.jpg', '76_485.jpg', '76_495.jpg', '76_496.jpg', '76_546.jpg', '76_547.jpg', '76_564.jpg', '76_586.jpg']

2025-03-31 10:19:42 [ERROR] 
2025-03-31 10:19:42 [INFO] Done removing outliers
2025-03-31 10:19:42 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:19:42 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 10:19:43 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.5112314224243164, 0.5124539659078893, 0.5531376004219055
2025-03-31 10:19:43 [INFO] Median vs. average vs. max of kep

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:19:49 [INFO] 2025-03-31 10:19:47 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:19:47 [INFO] Round 1: Excluded 7 outliers
2025-03-31 10:19:47 [INFO] Round 1: Average score: 0.9965685606002808
2025-03-31 10:19:47 [INFO] Excluded images: ['77_172.jpg', '77_173.jpg', '77_174.jpg', '77_570.jpg', '77_571.jpg', '77_573.jpg', '77_579.jpg']

2025-03-31 10:19:49 [ERROR] 
2025-03-31 10:19:49 [INFO] Done removing outliers
2025-03-31 10:19:49 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:19:49 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 10:19:50 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.46236729621887207, 0.4631303603699944, 0.5022352337837219
2025-03-31 10:19:50 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5010455250740051, 0.5012234896421432, 0.5022352337837219
2025-03-31 10:19:50 [INFO] Tracklet 67 is legible.
2025-03-31 10:19:50 [INFO] Saving legible_tracklets to: c:\Use

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:19:54 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.38740573823451996, 0.3886790183709941, 0.4184151291847229
2025-03-31 10:19:54 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.0, 0.0, 0.0
2025-03-31 10:19:54 [INFO] Tracklet 73 is illegible.
2025-03-31 10:19:54 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\73\legible_results.json
2025-03-31 10:19:54 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\73\legible_results.json
2025-03-31 10:19:54 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\73\illegible_results.json
2025-03-31 10:19:54 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jer

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:20:01 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.5367279648780823, 0.53488815825614, 0.5613462328910828
2025-03-31 10:20:01 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5368131995201111, 0.5349392997825163, 0.5613462328910828
2025-03-31 10:20:01 [INFO] Tracklet 70 is legible.
2025-03-31 10:20:01 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\70\legible_results.json
2025-03-31 10:20:01 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\70\legible_results.json
2025-03-31 10:20:01 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\70\illegible_results.json
2025-03-31 10:20:01 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jers

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:20:09 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.5564728379249573, 0.5549146519521627, 0.5804013013839722
2025-03-31 10:20:09 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5564728379249573, 0.5549146519521627, 0.5804013013839722
2025-03-31 10:20:09 [INFO] Tracklet 77 is legible.
2025-03-31 10:20:09 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\77\legible_results.json
2025-03-31 10:20:09 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\77\legible_results.json
2025-03-31 10:20:09 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\77\illegible_results.json
2025-03-31 10:20:09 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Je

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:20:12 [INFO] 2025-03-31 10:20:10 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:20:10 [INFO] Round 1: Excluded 23 outliers
2025-03-31 10:20:10 [INFO] Round 1: Average score: 0.9809175729751587
2025-03-31 10:20:10 [INFO] Excluded images: ['80_205.jpg', '80_206.jpg', '80_207.jpg', '80_326.jpg', '80_570.jpg', '80_571.jpg', '80_572.jpg', '80_573.jpg', '80_579.jpg', '80_580.jpg', '80_581.jpg', '80_583.jpg', '80_587.jpg', '80_588.jpg', '80_589.jpg', '80_591.jpg', '80_592.jpg', '80_593.jpg', '80_595.jpg', '80_596.jpg', '80_597.jpg', '80_598.jpg', '80_605.jpg']

2025-03-31 10:20:12 [ERROR] 
2025-03-31 10:20:12 [INFO] Done removing outliers
2025-03-31 10:20:12 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:20:12 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 10:20:15 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.5579037070274353, 0.5568955048889099, 0.6014201641082764
2025-03-31 10:20:15 [INFO] Media

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:20:18 [INFO] 2025-03-31 10:20:16 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:20:17 [INFO] Round 1: Excluded 41 outliers
2025-03-31 10:20:17 [INFO] Round 1: Average score: 0.9878318905830383
2025-03-31 10:20:17 [INFO] Excluded images: ['81_163.jpg', '81_164.jpg', '81_176.jpg', '81_187.jpg', '81_188.jpg', '81_189.jpg', '81_200.jpg', '81_201.jpg', '81_202.jpg', '81_204.jpg', '81_205.jpg', '81_206.jpg', '81_207.jpg', '81_208.jpg', '81_209.jpg', '81_210.jpg', '81_211.jpg', '81_453.jpg', '81_609.jpg', '81_610.jpg', '81_676.jpg', '81_677.jpg', '81_680.jpg', '81_681.jpg', '81_682.jpg', '81_683.jpg', '81_684.jpg', '81_687.jpg', '81_688.jpg', '81_692.jpg', '81_696.jpg', '81_697.jpg', '81_698.jpg', '81_699.jpg', '81_704.jpg', '81_705.jpg', '81_706.jpg', '81_713.jpg', '81_716.jpg', '81_743.jpg', '81_750.jpg']

2025-03-31 10:20:18 [ERROR] 
2025-03-31 10:20:18 [INFO] Done removing outliers
2025-03-31 10:20:18 [INFO] Running model chain on preprocessed image(s).
2025-03-3

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:20:21 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.5163969993591309, 0.5169421158592153, 0.5396838784217834
2025-03-31 10:20:21 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5164221227169037, 0.5169818239340643, 0.5396838784217834
2025-03-31 10:20:21 [INFO] Tracklet 79 is legible.
2025-03-31 10:20:21 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\79\legible_results.json
2025-03-31 10:20:21 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\79\legible_results.json
2025-03-31 10:20:21 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\79\illegible_results.json
2025-03-31 10:20:21 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Je

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:20:29 [INFO] 2025-03-31 10:20:27 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:20:27 [INFO] Round 1: Excluded 22 outliers
2025-03-31 10:20:27 [INFO] Round 1: Average score: 0.9911590814590454
2025-03-31 10:20:27 [INFO] Excluded images: ['84_181.jpg', '84_182.jpg', '84_185.jpg', '84_186.jpg', '84_187.jpg', '84_206.jpg', '84_207.jpg', '84_256.jpg', '84_257.jpg', '84_302.jpg', '84_303.jpg', '84_328.jpg', '84_329.jpg', '84_579.jpg', '84_580.jpg', '84_581.jpg', '84_582.jpg', '84_583.jpg', '84_587.jpg', '84_589.jpg', '84_598.jpg', '84_599.jpg']

2025-03-31 10:20:29 [ERROR] 
2025-03-31 10:20:29 [INFO] Done removing outliers
2025-03-31 10:20:29 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:20:29 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 10:20:30 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.5719288289546967, 0.5719242644309998, 0.623291552066803
2025-03-31 10:20:30 [INFO] Median vs. average v

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:20:33 [INFO] 2025-03-31 10:20:32 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:20:32 [INFO] Round 1: Excluded 47 outliers
2025-03-31 10:20:32 [INFO] Round 1: Average score: 0.9812021851539612
2025-03-31 10:20:32 [INFO] Excluded images: ['85_188.jpg', '85_189.jpg', '85_190.jpg', '85_191.jpg', '85_192.jpg', '85_193.jpg', '85_194.jpg', '85_195.jpg', '85_196.jpg', '85_197.jpg', '85_198.jpg', '85_199.jpg', '85_200.jpg', '85_201.jpg', '85_202.jpg', '85_203.jpg', '85_204.jpg', '85_205.jpg', '85_206.jpg', '85_207.jpg', '85_208.jpg', '85_209.jpg', '85_210.jpg', '85_211.jpg', '85_212.jpg', '85_213.jpg', '85_214.jpg', '85_215.jpg', '85_216.jpg', '85_217.jpg', '85_218.jpg', '85_246.jpg', '85_247.jpg', '85_248.jpg', '85_249.jpg', '85_250.jpg', '85_251.jpg', '85_252.jpg', '85_253.jpg', '85_254.jpg', '85_256.jpg', '85_259.jpg', '85_260.jpg', '85_583.jpg', '85_584.jpg', '85_586.jpg', '85_603.jpg']

2025-03-31 10:20:33 [ERROR] 
2025-03-31 10:20:33 [INFO] Done removing outlier

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:20:38 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.4760735481977463, 0.47479524951295804, 0.5095895528793335
2025-03-31 10:20:38 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5018932223320007, 0.504112442334493, 0.5095895528793335
2025-03-31 10:20:38 [INFO] Tracklet 81 is legible.
2025-03-31 10:20:38 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\81\legible_results.json
2025-03-31 10:20:38 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\81\legible_results.json
2025-03-31 10:20:38 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\81\illegible_results.json
2025-03-31 10:20:38 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Je

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:20:43 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.46024399995803833, 0.4596625290265897, 0.5053156614303589
2025-03-31 10:20:43 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5053156614303589, 0.5053156614303589, 0.5053156614303589
2025-03-31 10:20:43 [INFO] Tracklet 82 is legible.
2025-03-31 10:20:43 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\82\legible_results.json
2025-03-31 10:20:43 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\82\legible_results.json
2025-03-31 10:20:43 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\82\illegible_results.json
2025-03-31 10:20:43 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\J

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:20:50 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.46334370970726013, 0.4618978982308049, 0.521320104598999
2025-03-31 10:20:50 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5049749612808228, 0.507506081036159, 0.521320104598999
2025-03-31 10:20:50 [INFO] Tracklet 84 is legible.
2025-03-31 10:20:50 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\84\legible_results.json
2025-03-31 10:20:50 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\84\legible_results.json
2025-03-31 10:20:50 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\84\illegible_results.json
2025-03-31 10:20:50 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jers

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:20:57 [INFO] 2025-03-31 10:20:56 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:20:56 [INFO] Round 1: Excluded 3 outliers
2025-03-31 10:20:56 [INFO] Round 1: Average score: 0.9950174689292908
2025-03-31 10:20:56 [INFO] Excluded images: ['83_176.jpg', '83_587.jpg', '83_610.jpg']

2025-03-31 10:20:57 [ERROR] 
2025-03-31 10:20:57 [INFO] Done removing outliers
2025-03-31 10:20:57 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:20:57 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 10:20:58 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.47200414538383484, 0.4736539957759617, 0.5200740098953247
2025-03-31 10:20:58 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5033028423786163, 0.5057102143764496, 0.5200740098953247
2025-03-31 10:20:58 [INFO] Saved features for tracklet with shape (269, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\pro

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`
Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:21:02 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.4477126896381378, 0.44843429896482234, 0.48248669505119324
2025-03-31 10:21:02 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.0, 0.0, 0.0
2025-03-31 10:21:02 [INFO] Tracklet 87 is illegible.
2025-03-31 10:21:02 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\87\legible_results.json
2025-03-31 10:21:02 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\87\legible_results.json
2025-03-31 10:21:02 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\87\illegible_results.json
2025-03-31 10:21:02 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\je

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:21:08 [INFO] 2025-03-31 10:21:07 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:21:07 [INFO] Round 1: Excluded 9 outliers
2025-03-31 10:21:07 [INFO] Round 1: Average score: 0.9914602041244507
2025-03-31 10:21:07 [INFO] Excluded images: ['91_215.jpg', '91_216.jpg', '91_217.jpg', '91_218.jpg', '91_219.jpg', '91_272.jpg', '91_589.jpg', '91_599.jpg', '91_604.jpg']

2025-03-31 10:21:08 [ERROR] 
2025-03-31 10:21:08 [INFO] Done removing outliers
2025-03-31 10:21:08 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:21:08 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 10:21:09 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.46315184235572815, 0.46524600982666015, 0.5048914551734924
2025-03-31 10:21:09 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5012081265449524, 0.5021095077196757, 0.5048914551734924
2025-03-31 10:21:09 [INFO] Tracklet 88 is legible.
2025-03-31 10:21:09 [INFO] Saving

Processing tracklets (CUDA + CPU):   0%|          | 0/32 [00:00<?, ?it/s]

2025-03-31 10:22:26 [INFO] Creating placeholder data files for Soccer Ball Filter.


Processing Batch Tracklets (96-128):   0%|          | 0/32 [00:00<?, ?it/s]

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:22:29 [INFO] Saved features for tracklet with shape (416, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\96\features.npy
2025-03-31 10:22:29 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:22:31 [INFO] Saved features for tracklet with shape (622, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\98\features.npy
2025-03-31 10:22:31 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:22:33 [INFO] Saved features for tracklet with shape (750, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\100\features.npy
2025-03-31 10:22:33 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:22:36 [INFO] Saved features for tracklet with shape (553, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\103\features.npy
2025-03-31 10:22:36 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:22:38 [INFO] Saved features for tracklet with shape (624, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\104\features.npy
2025-03-31 10:22:38 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:22:39 [INFO] 2025-03-31 10:22:38 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:22:38 [INFO] Round 1: Excluded 23 outliers
2025-03-31 10:22:38 [INFO] Round 1: Average score: 0.9900973439216614
2025-03-31 10:22:38 [INFO] Excluded images: ['96_172.jpg', '96_507.jpg', '96_508.jpg', '96_510.jpg', '96_534.jpg', '96_535.jpg', '96_547.jpg', '96_560.jpg', '96_582.jpg', '96_583.jpg', '96_584.jpg', '96_585.jpg', '96_586.jpg', '96_587.jpg', '96_588.jpg', '96_589.jpg', '96_590.jpg', '96_591.jpg', '96_592.jpg', '96_596.jpg', '96_597.jpg', '96_598.jpg', '96_750.jpg']

2025-03-31 10:22:39 [ERROR] 
2025-03-31 10:22:39 [INFO] Done removing outliers
2025-03-31 10:22:39 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:22:39 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 10:22:42 [INFO] 2025-03-31 10:22:40 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:22:40 [INFO] Round 1: Excluded 16 outliers
2025-03-31 10:22:40 [INFO

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:22:46 [INFO] 2025-03-31 10:22:44 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:22:44 [INFO] Round 1: Excluded 11 outliers
2025-03-31 10:22:44 [INFO] Round 1: Average score: 0.9954562187194824
2025-03-31 10:22:44 [INFO] Excluded images: ['100_217.jpg', '100_218.jpg', '100_219.jpg', '100_220.jpg', '100_221.jpg', '100_222.jpg', '100_223.jpg', '100_288.jpg', '100_293.jpg', '100_389.jpg', '100_390.jpg']

2025-03-31 10:22:46 [ERROR] 
2025-03-31 10:22:46 [INFO] Done removing outliers
2025-03-31 10:22:46 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:22:46 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 10:22:48 [INFO] Saved features for tracklet with shape (378, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\101\features.npy
2025-03-31 10:22:48 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:22:49 [INFO] 2025-03-31 10:22:48 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:22:48 [INFO] Round 1: Excluded 11 outliers
2025-03-31 10:22:48 [INFO] Round 1: Average score: 0.9956412315368652
2025-03-31 10:22:48 [INFO] Excluded images: ['103_302.jpg', '103_303.jpg', '103_357.jpg', '103_358.jpg', '103_359.jpg', '103_434.jpg', '103_435.jpg', '103_436.jpg', '103_437.jpg', '103_445.jpg', '103_447.jpg']

2025-03-31 10:22:49 [ERROR] 
2025-03-31 10:22:49 [INFO] Done removing outliers
2025-03-31 10:22:49 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:22:49 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 10:22:54 [INFO] 2025-03-31 10:22:52 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:22:52 [INFO] Round 1: Excluded 11 outliers
2025-03-31 10:22:52 [INFO] Round 1: Average score: 0.9965878129005432
2025-03-31 10:22:52 [INFO] Excluded images: ['104_158.jpg', '104_159.jpg', '104_160.jpg', '104_168.jpg', '104_18

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:22:56 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.466007798910141, 0.46696491171688825, 0.5039564371109009
2025-03-31 10:22:56 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5039564371109009, 0.5039564371109009, 0.5039564371109009
2025-03-31 10:22:56 [INFO] Tracklet 96 is legible.
2025-03-31 10:22:56 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\96\legible_results.json
2025-03-31 10:22:56 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\96\legible_results.json
2025-03-31 10:22:56 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\96\illegible_results.json
2025-03-31 10:22:56 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Je

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:23:03 [INFO] Saved features for tracklet with shape (584, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\99\features.npy
2025-03-31 10:23:03 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file
2025-03-31 10:23:03 [INFO] 2025-03-31 10:23:02 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:23:02 [INFO] Round 1: Excluded 8 outliers
2025-03-31 10:23:02 [INFO] Round 1: Average score: 0.9956125617027283
2025-03-31 10:23:02 [INFO] Excluded images: ['101_297.jpg', '101_298.jpg', '101_299.jpg', '101_300.jpg', '101_301.jpg', '101_302.jpg', '101_303.jpg', '101_639.jpg']

2025-03-31 10:23:03 [ERROR] 
2025-03-31 10:23:03 [INFO] Done removing outliers
2025-03-31 10:23:03 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:23:03 [INFO] Classifying legibility of image(s) using pre-trained model.


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:23:07 [INFO] Saved features for tracklet with shape (547, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\106\features.npy
2025-03-31 10:23:07 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file
2025-03-31 10:23:11 [INFO] 2025-03-31 10:23:09 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:23:09 [INFO] Round 1: Excluded 16 outliers
2025-03-31 10:23:09 [INFO] Round 1: Average score: 0.9940771460533142
2025-03-31 10:23:09 [INFO] Excluded images: ['105_155.jpg', '105_156.jpg', '105_157.jpg', '105_158.jpg', '105_159.jpg', '105_160.jpg', '105_161.jpg', '105_162.jpg', '105_164.jpg', '105_296.jpg', '105_297.jpg', '105_605.jpg', '105_655.jpg', '105_656.jpg', '105_657.jpg', '105_658.jpg']

2025-03-31 10:23:11 [ERROR] 
2025-03-31 10:23:11 [INFO] Done removing outliers
2025-03-31 10:23:11 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:23:11 [INFO]

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:23:20 [INFO] 2025-03-31 10:23:18 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:23:18 [INFO] Round 1: Excluded 7 outliers
2025-03-31 10:23:18 [INFO] Round 1: Average score: 0.9965168237686157
2025-03-31 10:23:18 [INFO] Excluded images: ['99_249.jpg', '99_250.jpg', '99_251.jpg', '99_252.jpg', '99_253.jpg', '99_581.jpg', '99_584.jpg']

2025-03-31 10:23:20 [ERROR] 
2025-03-31 10:23:20 [INFO] Done removing outliers
2025-03-31 10:23:20 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:23:20 [INFO] Classifying legibility of image(s) using pre-trained model.
2025-03-31 10:23:24 [INFO] 2025-03-31 10:23:23 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:23:23 [INFO] Round 1: Excluded 23 outliers
2025-03-31 10:23:23 [INFO] Round 1: Average score: 0.9912558197975159
2025-03-31 10:23:23 [INFO] Excluded images: ['106_137.jpg', '106_138.jpg', '106_139.jpg', '106_140.jpg', '106_141.jpg', '106_142.jpg', '106_143.jpg', '106_145.jpg', '106_146.jpg', 

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:23:36 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.49202418327331543, 0.49292450118935965, 0.52999347448349
2025-03-31 10:23:36 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5072200894355774, 0.5089269718491888, 0.52999347448349
2025-03-31 10:23:36 [INFO] Tracklet 100 is legible.
2025-03-31 10:23:36 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\100\legible_results.json
2025-03-31 10:23:36 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\100\legible_results.json
2025-03-31 10:23:36 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\100\illegible_results.json
2025-03-31 10:23:36 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:23:37 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.5263054668903351, 0.5271216563303093, 0.5601808428764343
2025-03-31 10:23:37 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5263579189777374, 0.5272290324723279, 0.5601808428764343
2025-03-31 10:23:37 [INFO] Tracklet 103 is legible.
2025-03-31 10:23:37 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\103\legible_results.json
2025-03-31 10:23:37 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\103\legible_results.json
2025-03-31 10:23:37 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\103\illegible_results.json
2025-03-31 10:23:37 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Deskto

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:23:43 [INFO] Tracklet 104 is legible.
2025-03-31 10:23:43 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\104\legible_results.json
2025-03-31 10:23:43 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\104\legible_results.json
2025-03-31 10:23:43 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\104\illegible_results.json
2025-03-31 10:23:43 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\104\illegible_results.json
2025-03-31 10:23:43 [INFO] Legibility classification complete.
2025-03-31 10:23:43 [INFO] Processed tracklet: 104
2025-03-31 10:23:46 [INFO] Median vs. average vs. max of raw out

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:23:51 [INFO] 2025-03-31 10:23:50 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:23:50 [INFO] Round 1: Excluded 41 outliers
2025-03-31 10:23:50 [INFO] Round 1: Average score: 0.9881247878074646
2025-03-31 10:23:50 [INFO] Excluded images: ['108_173.jpg', '108_174.jpg', '108_175.jpg', '108_176.jpg', '108_177.jpg', '108_178.jpg', '108_200.jpg', '108_208.jpg', '108_209.jpg', '108_210.jpg', '108_211.jpg', '108_212.jpg', '108_213.jpg', '108_305.jpg', '108_421.jpg', '108_571.jpg', '108_572.jpg', '108_573.jpg', '108_575.jpg', '108_576.jpg', '108_577.jpg', '108_578.jpg', '108_579.jpg', '108_580.jpg', '108_581.jpg', '108_582.jpg', '108_583.jpg', '108_584.jpg', '108_585.jpg', '108_586.jpg', '108_587.jpg', '108_588.jpg', '108_599.jpg', '108_711.jpg', '108_712.jpg', '108_713.jpg', '108_714.jpg', '108_715.jpg', '108_716.jpg', '108_718.jpg', '108_729.jpg']

2025-03-31 10:23:51 [ERROR] 
2025-03-31 10:23:51 [INFO] Done removing outliers
2025-03-31 10:23:51 [INFO] Running model 

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:23:55 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.5490998327732086, 0.54823864718615, 0.5652661323547363
2025-03-31 10:23:55 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5490998327732086, 0.54823864718615, 0.5652661323547363
2025-03-31 10:23:55 [INFO] Tracklet 105 is legible.
2025-03-31 10:23:55 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\105\legible_results.json
2025-03-31 10:23:55 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\105\legible_results.json
2025-03-31 10:23:55 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\105\illegible_results.json
2025-03-31 10:23:55 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Je

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:23:58 [INFO] 2025-03-31 10:23:57 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:23:57 [INFO] Round 1: Excluded 44 outliers
2025-03-31 10:23:57 [INFO] Round 1: Average score: 0.9891157150268555
2025-03-31 10:23:57 [INFO] Excluded images: ['109_126.jpg', '109_127.jpg', '109_128.jpg', '109_129.jpg', '109_131.jpg', '109_132.jpg', '109_133.jpg', '109_134.jpg', '109_135.jpg', '109_136.jpg', '109_137.jpg', '109_138.jpg', '109_139.jpg', '109_174.jpg', '109_177.jpg', '109_178.jpg', '109_179.jpg', '109_181.jpg', '109_182.jpg', '109_199.jpg', '109_200.jpg', '109_201.jpg', '109_203.jpg', '109_204.jpg', '109_214.jpg', '109_215.jpg', '109_216.jpg', '109_219.jpg', '109_220.jpg', '109_221.jpg', '109_222.jpg', '109_223.jpg', '109_224.jpg', '109_225.jpg', '109_227.jpg', '109_239.jpg', '109_432.jpg', '109_497.jpg', '109_498.jpg', '109_499.jpg', '109_500.jpg', '109_501.jpg', '109_502.jpg', '109_503.jpg']

2025-03-31 10:23:58 [ERROR] 
2025-03-31 10:23:58 [INFO] Done removing outli

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:24:06 [INFO] Median vs. average vs. max of raw outputs (thresh=0.5): 0.5654971599578857, 0.5655879617976977, 0.5864173173904419
2025-03-31 10:24:06 [INFO] Median vs. average vs. max of kept outputs (thresh=0.5): 0.5654971599578857, 0.5655879617976977, 0.5864173173904419
2025-03-31 10:24:06 [INFO] Tracklet 99 is legible.
2025-03-31 10:24:06 [INFO] Saving legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\99\legible_results.json
2025-03-31 10:24:06 [INFO] Saved legible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\99\legible_results.json
2025-03-31 10:24:06 [INFO] Saving illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\99\illegible_results.json
2025-03-31 10:24:06 [INFO] Saved illegible_tracklets to: c:\Users\colin\OneDrive\Desktop\Je

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:24:12 [INFO] Saved features for tracklet with shape (491, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\115\features.npy
2025-03-31 10:24:12 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:24:13 [INFO] 2025-03-31 10:24:12 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:24:12 [INFO] Round 1: Excluded 35 outliers
2025-03-31 10:24:12 [INFO] Round 1: Average score: 0.990524411201477
2025-03-31 10:24:12 [INFO] Excluded images: ['112_131.jpg', '112_132.jpg', '112_133.jpg', '112_134.jpg', '112_135.jpg', '112_136.jpg', '112_137.jpg', '112_138.jpg', '112_139.jpg', '112_140.jpg', '112_141.jpg', '112_143.jpg', '112_144.jpg', '112_145.jpg', '112_146.jpg', '112_147.jpg', '112_151.jpg', '112_206.jpg', '112_207.jpg', '112_208.jpg', '112_220.jpg', '112_229.jpg', '112_231.jpg', '112_258.jpg', '112_267.jpg', '112_268.jpg', '112_269.jpg', '112_270.jpg', '112_271.jpg', '112_272.jpg', '112_273.jpg', '112_274.jpg', '112_276.jpg', '112_277.jpg', '112_278.jpg']

2025-03-31 10:24:13 [ERROR] 
2025-03-31 10:24:13 [INFO] Done removing outliers
2025-03-31 10:24:13 [INFO] Running model chain on preprocessed image(s).
2025-03-31 10:24:13 [INFO] Classifying legibility of image(

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:24:29 [INFO] Saved features for tracklet with shape (355, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\117\features.npy
2025-03-31 10:24:29 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file
2025-03-31 10:24:29 [INFO] 2025-03-31 10:24:28 [INFO] Threshold for scores: 0.97, rounds: 1
2025-03-31 10:24:28 [INFO] Round 1: Excluded 35 outliers
2025-03-31 10:24:28 [INFO] Round 1: Average score: 0.9888723492622375
2025-03-31 10:24:28 [INFO] Excluded images: ['115_105.jpg', '115_106.jpg', '115_165.jpg', '115_170.jpg', '115_180.jpg', '115_192.jpg', '115_452.jpg', '115_453.jpg', '115_534.jpg', '115_535.jpg', '115_536.jpg', '115_537.jpg', '115_538.jpg', '115_573.jpg', '115_578.jpg', '115_579.jpg', '115_580.jpg', '115_581.jpg', '115_582.jpg', '115_583.jpg', '115_584.jpg', '115_585.jpg', '115_586.jpg', '115_587.jpg', '115_588.jpg', '115_589.jpg', '115_707.jpg', '115_709.jpg',

Lightning automatically upgraded your loaded checkpoint from v1.1.4 to v2.5.0.post0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\pre_trained_models\reid\dukemtmcreid_resnet50_256_128_epoch_120.ckpt`


2025-03-31 10:24:43 [INFO] Saved features for tracklet with shape (350, 2048) to c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\challenge\118\features.npy
2025-03-31 10:24:43 [INFO] Identifying and removing outliers by calling gaussian_outliers.py on feature file


## Benchmark (current pipeline metrics)
### Results
- 2025-03-29 14:44:05 [INFO] Correct 1051 out of 1210. Accuracy 86.85950413223141%.
- 2025-03-29 14:44:05 [INFO] TP=766, TN=285, FP=70, FN=89
- 2025-03-29 14:44:05 [INFO] Precision=0.916267942583732, Recall=0.895906432748538
- 2025-03-29 14:44:05 [INFO] F1=0.9059727971614429
- 2025-03-29 14:44:05 [INFO] Saved failed legibility cases to:  c:\Users\colin\OneDrive\Desktop\Jersey-Number-Recognition\data\SoccerNet\jersey-2023\processed_data\test\common_data\failed_legibility_cases.json

### Analysis
legible_but_marked_illegible: 89 cases
109, 88, 34, 151, 19, 68, 67, 128, 57, 178, 189, 196, 214, 242, 258, 271, 304, 327, 320, 352, 375, 378, 414, 416, 439, 425, 451, 453, 454, 466, 499, 511, 513, 526, 528, 567, 576, 594, 619, 623, 646, 687, 682, 709, 711, 717, 739, 758, 767, 795, 788, 840, 843, 832, 841, 852, 847, 851, 856, 850, 868, 866, 871, 901, 902, 904, 900, 907, 928, 939, 944, 961, 984, 983, 1005, 1039, 1064, 1063, 1077, 1091, 1101, 1137, 1152, 1153, 1160, 1167, 1157, 1165, 1206

illegible_but_marked_legible: 70 cases
168, 167, 162, 97, 39, 87, 89, 134, 61, 172, 164, 199, 215, 260, 267, 269, 281, 276, 288, 295, 317, 325, 340, 345, 374, 394, 388, 386, 391, 406, 408, 434, 437, 459, 471, 514, 532, 558, 683, 748, 770, 790, 802, 827, 861, 869, 879, 897, 895, 909, 964, 981, 999, 1023, 1031, 1025, 1033, 1030, 1055, 1088, 1087, 1169, 1175, 1174, 1173, 1183, 1170, 1179, 1184, 1185

- Total misclassifications: 159
- False negatives (legible marked as illegible): 89
- False positives (illegible marked as legible): 70
- Percentage of false negatives: 7.36%
- Percentage of false positives: 5.79%
- Combined percentage of errors as a percentage of total cases: 13.14%

In [ ]:
import json
failed_cases = f"{DataPaths.COMMON_PROCESSED_OUTPUT_DATA_CHALLENGE.value}/failed_legibility_cases.json"

# Load this JSON file
with open(failed_cases, 'r') as f:
    failed_cases_dict = json.load(f)
    # Print the every key and its associated value (value will be a list so need to make that printable with ",".join):
    for key, value in failed_cases_dict.items():
        print(f"{key}: {len(value)} cases")
        if len(value) > 0:
            print(f"{', '.join(value)}")
        print()
    
    # Summary statistics
    legible_but_marked_illegible = len(failed_cases_dict.get("legible_but_marked_illegible", []))
    illegible_but_marked_legible = len(failed_cases_dict.get("illegible_but_marked_legible", []))
    total_errors = legible_but_marked_illegible + illegible_but_marked_legible
    
    print(f"Total misclassifications: {total_errors}")
    print(f"False negatives (legible marked as illegible): {legible_but_marked_illegible}")
    print(f"False positives (illegible marked as legible): {illegible_but_marked_legible}")
    print(f"Percentage of false negatives: {legible_but_marked_illegible / 1210 * 100:.2f}%")
    print(f"Percentage of false positives: {illegible_but_marked_legible / 1210 * 100:.2f}%")
    print(f"Combined percentage of errors as a percentage of total cases: {total_errors / 1210 * 100:.2f}%")

In [ ]:
# import timm
# import torch
# from torch import nn

# model = timm.create_model("timm/vit_base_patch16_224.orig_in21k_ft_in1k", pretrained=False)
# model.head = nn.Linear(model.head.in_features, 10)
# model.load_state_dict(
#     torch.hub.load_state_dict_from_url(
#         "https://huggingface.co/edadaltocg/vit_base_patch16_224_in21k_ft_svhn/resolve/main/pytorch_model.bin",
#         map_location="cpu",
#         file_name="vit_base_patch16_224_in21k_ft_svhn.pth",
#     )
# )